services/QuillLMS/app/controllers/admins_controller.rb

```diff
@@ -43,6 +53,72 @@ def sign_in_account_settings
     redirect_to teachers_my_account_path
   end
 
+  def resend_login_details
+    @teacher.refresh_token!
+    ExpirePasswordTokenWorker.perform_in(30.days, @teacher.id)
+
+    if params[:role] == 'admin'
+      AdminDashboard::AdminAccountCreatedEmailWorker.perform_async(@teacher.id, current_user.id, params[:school_id], true)
+    else
+      AdminDashboard::TeacherAccountCreatedEmailWorker.perform_async(@teacher.id, current_user.id, params[:school_id], true)
+    end
+
+    render json: {message: t('admin.resend_login_details')}, status: 200
+  end
+
+  def remove_as_admin
+    SchoolsAdmins.find_by(user_id: params[:id], school_id: params[:school_id]).destroy!
+    reset_admin_users_cache
+    render json: {message: t('admin.remove_admin')}, status: 200
+  end
+
+  def make_admin
+    SchoolsAdmins.create!(user_id: params[:id], school_id: params[:school_id])
+    AdminDashboard::MadeSchoolAdminEmailWorker.perform_async(params[:id], current_user.id, params[:school_id])
+    reset_admin_users_cache
+    render json: {message: t('admin.make_admin')}, status: 200
+  end
+
+  def unlink_from_school
+    @teacher&.unlink
+    reset_admin_users_cache
+    render json: {message: t('admin.unlink_teacher_from_school')}, status: 200
+  end
+
+  def create_and_link_accounts
+    @school = School.find_by(id: params[:school_id])
+
+    if SchoolsAdmins.find_by(school: @school, user: current_user).nil?
+      render json: {errors: 'Something went wrong. If this problem persists, please contact us at hello@quill.org'}, status: 422
+      return
+    end
+
+    @teacher = User.find_by(email: teacher_params[:email])
+
+    @is_admin = teacher_params[:role] == 'admin'
+    teacher_params.delete(:role)
+
+    if @teacher
+      if @is_admin
+        handle_existing_user_submitted_as_admin
+      else
+        handle_existing_user_submitted_as_teacher
+      end
+    else
+      # Create a new teacher, and automatically join them to the school.
```

Comment:
I would move this comment into the `handle_new_user` method since it seems to have more functionality now.